pkg/cli/admin/upgrade/recommend/recommend.go

```diff
@@ -145,82 +138,138 @@ func (o *options) Run(ctx context.Context) error {
-			fmt.Fprintf(o.ErrOut, "warning: Cannot refresh available updates:\n  Reason: %s\n  Message: %s\n\n", c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  "))
+
+		if minorBuckets := majorMinorBuckets[version.Major]; minorBuckets == nil {
+			majorMinorBuckets[version.Major] = make(map[uint64][]configv1.ConditionalUpdate, 0)
 		}
-	} else {
-		if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
-			fmt.Fprintf(o.ErrOut, "warning: Cannot display available updates:\n  Reason: %s\n  Message: %s\n\n", c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  "))
-		} else {
-			fmt.Fprintf(o.Out, "No updates available. You may still upgrade to a specific release image with --to-image or wait for new updates to be available.\n")
+
+		if minorBucket := majorMinorBuckets[version.Major][version.Minor]; minorBucket == nil {
+			majorMinorBuckets[version.Major][version.Minor] = []configv1.ConditionalUpdate{}
 		}
+
+		majorMinorBuckets[version.Major][version.Minor] = append(majorMinorBuckets[version.Major][version.Minor], cv.Status.ConditionalUpdates[i])
 	}
 
-	if o.IncludeNotRecommended {
-		if containsNotRecommendedUpdate(cv.Status.ConditionalUpdates) {
-			sortConditionalUpdatesBySemanticVersions(cv.Status.ConditionalUpdates)
-			fmt.Fprintf(o.Out, "\nUpdates with known issues:\n")
-			for _, update := range cv.Status.ConditionalUpdates {
-				if c := findCondition(update.Conditions, "Recommended"); c != nil && c.Status != metav1.ConditionTrue {
-					fmt.Fprintf(o.Out, "\n  Version: %s\n  Image: %s\n", update.Release.Version, update.Release.Image)
-					fmt.Fprintf(o.Out, "  Reason: %s\n  Message: %s\n", c.Reason, strings.ReplaceAll(strings.TrimSpace(c.Message), "\n", "\n  "))
-				}
-			}
-		} else {
-			fmt.Fprintf(o.Out, "\nNo updates which are not recommended based on your cluster configuration are available.\n")
-		}
-	} else if containsNotRecommendedUpdate(cv.Status.ConditionalUpdates) {
-		qualifier := ""
-		for _, upgrade := range cv.Status.ConditionalUpdates {
-			if c := findCondition(upgrade.Conditions, "Recommended"); c != nil && c.Status != metav1.ConditionTrue && c.Status != metav1.ConditionFalse {
-				qualifier = fmt.Sprintf(", or where the recommended status is %q,", c.Status)
+	for i, update := range cv.Status.AvailableUpdates {
+		found := false
+		for _, conditionalUpdate := range cv.Status.ConditionalUpdates {
+			if conditionalUpdate.Release.Image == update.Image {
+				found = true
 				break
 			}
 		}
-		fmt.Fprintf(o.Out, "\nAdditional updates which are not recommended%s for your cluster configuration are available, to view those re-run the command with --include-not-recommended.\n", qualifier)
-	}
+		if found {
+			continue
+		}
 
-	// TODO: print previous versions
+		version, err := semver.Parse(update.Version)
+		if err != nil {
+			fmt.Fprintf(o.ErrOut, "warning: Cannot parse SemVer available update %q: %v", update.Version, err)
```

Comment:
`continue`

Reply:
I've pushed 46d8a1c096dc -> 07d7b03745e5b3, which adds this `continue`.